google/cloud/security/inventory/inventory_loader.py

```diff
@@ -257,15 +298,27 @@ def main(argv):
          'status': ''},
     ]
 
+    if configs.get('inventory_groups'):
+        if _should_inventory_google_groups(configs):
```

Comment:
nit: I feel a better place for this validation might be inside the load_groups_pipeline module, and raise LoadDataPipelineError if it can not run.

Here in the loader, it should be checking if the pipeline needs to run or not.  Whether it can or not should be inside the load_groups_pipeline module.